Login popup

sowmiyasubramaniyan opened the pull request:
fixes #1706 

Create a popup for login and register with the minimum input field

sandesh556 commented on app/Http/Controllers/Auth/RegisterController.php:
```diff
@@ -57,35 +58,40 @@ public function postRegister(ProfileRequest $request, User $user)
 
             return response()->json($emailMobileStatusResponse);
         } catch (\Exception $ex) {
+            dd($ex);
```
Remove dd from here.

sandesh556 commented on app/Http/Requests/User/ProfileRequest.php:
```diff
@@ -66,17 +62,23 @@ public function rules()
         }
 
-        if ($this->segment(1) == 'auth') {
+        if ($this->segment(1) == 'auth' && $value == '0') {
```
if $value is boolean no need to do == it will be already considered as true or false

sowmiyasubramaniyan replied:
Can you explain?

sandesh556 commented on app/Http/Requests/User/ProfileRequest.php:
```diff
@@ -85,8 +87,7 @@ public function rules()
         return[
-            'mobile_code.required'           => 'Enter Country code (mobile)',
-            'state.required_if'           => 'The state field is required when country is India.',
+            // 'mobile_code.required'           => 'Enter Country code (mobile)',
```
why did you comment out mobile_code?

sandesh556 commented on app/Http/Controllers/Auth/RegisterController.php:
```diff
@@ -57,29 +58,33 @@ public function postRegister(ProfileRequest $request, User $user)
 
+            $userData = 'Password:'.' '.$password;
+            $setting = Setting::find(1);
+            $mail = new \App\Http\Controllers\Common\PhpMailController();
```
Keep the imports always on top that is before the class, makes your code more readable

sandesh556 commented on resources/views/themes/default1/layouts/front/master.blade.php:
```diff
@@ -722,13 +753,63 @@ function removeItem(id) {
+    </script>
+<!--      <script src="{{asset('build/js/intlTelInput.js') }}"></script>
+ -->
+<!-- <script>
```
why is this commented out?

sandesh556 commented on resources/views/themes/default1/login-modal.blade.php:
```diff
@@ -0,0 +1,960 @@
+    </script>
+<!-- <script src="{{asset('build/js/intlTelInput.js') }}"></script>
+
+<script>
```
Why is this commented out?

sandesh556 commented on resources/views/themes/default1/login-modal.blade.php:
```diff
@@ -0,0 +1,960 @@
+<script type="text/javascript"
+        src="//www.googleadservices.com/pagead/conversion_async.js">
+</script>
+<!-- Facebook Pixel Code -->
```
Why is this commented?

sandesh556 commented on resources/views/themes/default1/login-modal.blade.php:
```diff
@@ -0,0 +1,960 @@
+                }
+            });
+
+            /*$(".next-step").click(function (e) {
```
Why is this commented?

sandesh556 commented on resources/views/themes/default1/login-modal.blade.php:
```diff
@@ -0,0 +1,960 @@
+<script src="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.13/js/select2.min.js"></script>
+<!-- <script>
+    var input = document.querySelector("#phonenum"),
+        errorMsg = document.querySelector("#error-msg"),
```
Why is this commented?

SarfrazS reviewed (changes requested):
Last commit
![image](https://user-images.githubusercontent.com/18435376/167830374-8b7cddcc-50e5-499a-93c5-914e00577494.png)

- [ ] Issue 1:- Free signup every time showing highlighted if we are in other tab.
![image](https://user-images.githubusercontent.com/18435376/167829567-736a32b0-9461-4c5b-9430-247f189387d8.png)
- [ ] Issue 2:- No Validation similarity on login and register page.
![image](https://user-images.githubusercontent.com/18435376/167829877-254b5608-7f49-4658-b8db-858f7f140105.png)
- [ ] Issue 3:- Enter all values and click on register success message not showing but user get registered.
- [ ] Register user not receiving registration email.
- [ ] No validation in mobile number.

sandesh556 commented:
And @SarfrazS have you confgured system mail?

mverma16 commented on app/Exceptions/Handler.php:
```diff
@@ -35,6 +35,7 @@ class Handler extends ExceptionHandler
      */
     public function report(Throwable $exception)
     {
+        dd($exception);
```
remove `dd()`, never leave the live dd statement anywhere in the code.

SarfrazS reviewed (changes requested):
Still few pointed issues are  not fixed yet please check it.
- [ ]  Issue 3:- Enter all values and click on register success message not showing but user got registered.

- [ ]  Register user not receiving registration email.
- [ ]  While registering user and country selecting Australia or any others then why picking state and city as bengaluru and karnataka
![image](https://user-images.githubusercontent.com/18435376/171583542-7ae5fae2-396d-461e-9eff-d70315417fe1.png)

Sheetal34 commented:
@sowmiyasubramaniyan Please can you give demo to QA Team, so that we can start testing ?

Sheetal34 commented:
As discussed with team, Implementation has not been completed yet. 
Note: If we are clicking on free signup tab, it's navigating to login page instead of register page.

Sheetal34 reviewed (changes requested):
**Tested with existing db:**

- After hitting url, how login page is showing ---- (Query)
![image](https://user-images.githubusercontent.com/92353368/174547201-47e5b981-555c-4a53-8517-5cb077b648d1.png)

- Register user by selecting country as Tuvalu and in mobile number select country code as India(Query)
![image](https://user-images.githubusercontent.com/92353368/174584509-6e416c53-0ae0-44a9-9e88-53fb9ed7fde0.png)
In admin panel click on all users, automatically selecting country code as Tuvalu 
![image](https://user-images.githubusercontent.com/92353368/174585101-67f54e1c-e540-42e4-b3ce-d3dbc9adabd5.png)
![image](https://user-images.githubusercontent.com/92353368/174585134-789cc7bf-e82c-4ade-90ca-30acb05a2076.png)

- While registering user we are not getting any successful message but user registered successfully and register page still loading
![Loading](https://user-images.githubusercontent.com/92353368/174586354-15bbbdf9-e61a-4d0c-8159-eda0673e9d4f.gif)

- In register page, add data and click on cross icon. Again click on Free signup tab still data is showing there. Now click on close tab, login page open automatically
![Sign in issue](https://user-images.githubusercontent.com/92353368/174587338-88c20e7b-feae-49e7-9388-215054996523.gif)

- In Register page, give first name and last name and click on submit, without giving any value in email throwing error please enter a valid email.
![Email issue in register page](https://user-images.githubusercontent.com/92353368/174587992-8576ac88-b104-409d-a3b4-228e40accb0c.gif)

- In Login page enter invalid email and submit. Didn't getting any validation for entering valid email. Now enter invalid email and invalid password also. Check in gif
![Login issue](https://user-images.githubusercontent.com/92353368/174589549-4f773c1a-327a-4bdb-883b-061bb99b4b27.gif)

- Registered user not getting any email for successful registration

Note: Testing not completed.

sowmiyasubramaniyan commented:
**After hitting url, how login page is showing ---- (Query)
**
Fixed

RafficMohammed commented on app/Http/Controllers/Auth/AuthController.php:
```diff
@@ -70,21 +70,32 @@ public function activate($token, AccountActivate $activate, Request $request, Us
             if ($ex->getCode() == 400) {
-                return redirect($url)->with('success', 'Email verification successful,
-                 Please login to access your account');
+                return redirect($url)->with(
```
Please Clean the code

sowmiyasubramaniyan replied:
done

sowmiyasubramaniyan commented:
**Register user by selecting the country as Tuvalu and in mobile number select country code as India(Query)
In the admin panel click on all users, automatically selecting the country code as Tuvalu**
  
    Country code will take depend on the country you are selecting

sowmiyasubramaniyan commented:
**While registering the user we are not getting any successful messages but the user registered successfully and the register page still loading**
  You need to configure your mail then you will get a success response and mail

sowmiyasubramaniyan commented:
**In register page, add data and click on cross icon. Again click on Free signup tab still data is showing there. Now click on close tab, login page open automatically**

    If we click the cross icon it will close the tab but won't delete the data and if we click the close tab it will refresh the page data will be deleted

sowmiyasubramaniyan commented:
**In Register page, give first name and last name and click on submit, without giving any value in email throwing error please enter a valid email**
  Fixed

sowmiyasubramaniyan commented:
**In Login page enter invalid email and submit. Didn't getting any validation for entering valid email. Now enter invalid email and invalid password also. Check in gif**
   Fixed

Sheetal34 reviewed (changes requested):
**Tested with existing db locally:**
Found issues there:

- [ ] No Validation similarity on login and register page ( Issue already pointed by sarfraz sir)
![image](https://user-images.githubusercontent.com/92353368/176163188-a665691f-d8f0-4389-9e8e-2288315e379e.png)
![image](https://user-images.githubusercontent.com/92353368/176163255-74377607-fb92-4c1d-9495-5112674e3505.png)
![image](https://user-images.githubusercontent.com/92353368/176163347-6b0e11a8-fd79-411d-9f90-2900a3a0bb81.png)

- [ ] In register page, we are giving invalid input in email field and try to submit, throwing error please enter an email
![image](https://user-images.githubusercontent.com/92353368/176163486-1aa8bc61-ffdb-4d4e-9c92-a2fd38c2dfd1.png)

sowmiyasubramaniyan commented:
Fixed @Sheetal34

Sheetal34 reviewed (approved):
**Tested with existing db locally:**
Mentioned issues are fixed

RafficMohammed commented on app/Http/Controllers/Auth/RegisterController.php:
```diff
@@ -54,32 +55,58 @@ public function postRegister(ProfileRequest $request, User $user)
-            $user->country = $request->input('country');
-            $user->password = $password;
+            $password = Str::random(20);
+            // $password = "Demo@123";
```
remove this line

priyadarshini2101 reviewed (changes requested):
- [ ]Issue 1: Reset password page issue and has no validation in login and register page.

![jul1reset](https://user-images.githubusercontent.com/91872895/176869439-8c4e2375-8556-4891-98cb-a8d5e5e44b31.gif)


https://docs.google.com/document/d/1dWysRYMvUQSB_GdxFf2iZY6q-27GUKlmW45YLnfo4Hg/edit

sowmiyasubramaniyan commented:
> * [ ]Issue 1: Reset password page issue and has no validation.
> 
> ![jul1reset](https://user-images.githubusercontent.com/91872895/176869439-8c4e2375-8556-4891-98cb-a8d5e5e44b31.gif) [ ](https://user-images.githubusercontent.com/91872895/176869439-8c4e2375-8556-4891-98cb-a8d5e5e44b31.gif) ![jul1reset](https://user-images.githubusercontent.com/91872895/176869439-8c4e2375-8556-4891-98cb-a8d5e5e44b31.gif) [ ](https://user-images.githubusercontent.com/91872895/176869439-8c4e2375-8556-4891-98cb-a8d5e5e44b31.gif)

Fixed

sowmiyasubramaniyan commented:
**Issue 1: Reset password page issue and has no validation.**
    Fixed

sowmiyasubramaniyan commented:
**Retested the pr found issues, updated, and explained to sowmya.

https://docs.google.com/document/d/1dWysRYMvUQSB_GdxFf2iZY6q-27GUKlmW45YLnfo4Hg/edit**
  Fixed all issues which are mentioned in the above document

Sheetal34 reviewed (changes requested):
**Tested in Somya's instance** 
https://developers.productdemourl.com/sowmya/public/

Found issues:

- [ ] Alignment issue in password fields 
![Screenshot (1295)](https://user-images.githubusercontent.com/92353368/179343450-ecdb760d-03c0-4262-82f7-6766f129ca58.png)

- [ ] Query:- I successfully reset my password, but if again I click on reset link, navigating to forgot password page-----------( Is it fine?)
![image](https://user-images.githubusercontent.com/92353368/179343543-542aba18-bf34-4344-b26e-b02898a209c9.png)
![image](https://user-images.githubusercontent.com/92353368/179343552-563b691c-4f0f-4326-b269-eb472ae05a26.png)

- [ ] Showing reset password page like this (Is it fine?) 
Why we are writing validation at top, we need to show validation under password fields 
![image](https://user-images.githubusercontent.com/92353368/179343575-4e01ac1d-11dc-40d4-af3d-7f47386d3421.png)

- [ ] Password field is not mandatory, but if we are trying to submit page without giving any input, throwing error password field is required
![image](https://user-images.githubusercontent.com/92353368/179343703-7fbf8c1c-37df-4914-81b8-c4bd6c06f5f9.png)

sowmiyasubramaniyan commented:
** Alignment issue in password fields**
Fixed

priyadarshini2101 reviewed (approved):
Tested the pr in server instance.

Pointed issues are fixed.

https://docs.google.com/document/d/1dWysRYMvUQSB_GdxFf2iZY6q-27GUKlmW45YLnfo4Hg/edit?usp=sharing

Isuues are there ,created seperate issues for the same.

https://github.com/ladybirdweb/agora-invoicing-community/issues/1866
https://github.com/ladybirdweb/agora-invoicing-community/issues/1868

Sheetal34 reviewed (approved):
Tested for mentioned issues. Working fine
